test/api/test_package.py

```diff
@@ -911,3 +915,68 @@ def _test_untag_package(
 
     assert_db_state(package_batch)
     assert_no_audit_log()
+
+
+def date_range_package_tag(package: PackageFactory, package_write_scope: ScopeFactory):
+    schema_uri = 'https://odp.saeon.ac.za/schema/tag/daterange'
+    return PackageTagFactory(
+        package=package,
+        tag=get_date_range_tag(ODPTagSchema.DATERANGE, schema_uri, ODPPackageTag.DATERANGE, package_write_scope),
+        data={
+            'start': '1990/01/01',
+            'end': '2000/01/01'
+        }
+    )
+
+
+def date_range_inc_package_tag(package: PackageFactory, package_write_scope: ScopeFactory):
+    schema_uri = 'https://odp.saeon.ac.za/schema/tag/daterangeinc'
+    return PackageTagFactory(
+        package=package,
+        tag=get_date_range_tag(ODPTagSchema.DATERANGEINC, schema_uri, ODPPackageTag.DATERANGEINC, package_write_scope),
+        data={
+            'end': ODPDateRangeIncType.CURRENT_DATE
+        }
+    )
+
+
+def get_date_range_tag(schema_id: str, schema_uri: str, tag_id: str, package_write_scope: ScopeFactory):
```

Comment:
The type hint for `package_write_scope` should be `Scope`. Note that this is true regardless of whether the scope was factory-created or retrieved from the factory session. Factories create actual ORM objects.